File: VB/XtraReport1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Drawing
Imports System.Drawing.Printing
Imports System.Collections
Imports System.ComponentModel
Imports DevExpress.XtraReports.UI

Namespace ShowProgressReflector
	Partial Public Class XtraReport1
		Inherits DevExpress.XtraReports.UI.XtraReport
		Public Sub New()
			InitializeComponent()
		End Sub

		Private i As Integer = 0

        Private Sub xrLabel2_BeforePrint(ByVal sender As Object, ByVal e As PrintEventArgs) _
        Handles xrLabel2.BeforePrint
            i += 1
            CType(sender, XRLabel).Text = "Row Number: " & i.ToString()
        End Sub

        Private Sub XtraReport1_BeforePrint(ByVal sender As Object, ByVal e As PrintEventArgs) _
        Handles MyBase.BeforePrint
            Me.DetailPrintCount = CInt(Fix(TotalRowCount.Value))
        End Sub

	End Class
End Namespace

File: CS/XtraReport1.cs
using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using DevExpress.XtraReports.UI;

namespace ShowProgressReflector {
    public partial class XtraReport1 : DevExpress.XtraReports.UI.XtraReport {
        public XtraReport1() {
            InitializeComponent();
        }

        int i = 0;

        private void xrLabel2_BeforePrint(object sender, System.Drawing.Printing.PrintEventArgs e) {
            i++;
            ((XRLabel)sender).Text = "Row Number: " + i.ToString();
        }

        private void XtraReport1_BeforePrint(object sender, System.Drawing.Printing.PrintEventArgs e) {
            i = 0;
            this.ReportPrintOptions.DetailCountOnEmptyDataSource = (int)TotalRowCount.Value;
        }

    }
}
